src/views/workflow-queries/workflow-queries-tile/workflow-queries-tile.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { Button, KIND, SIZE } from 'baseui/button';
+import { MdPlayArrow } from 'react-icons/md';
+
+import WorkflowQueriesTileInput from '../workflow-queries-tile-input/workflow-queries-tile-input';
+
+import { styled } from './workflow-queries-tile.styles';
+import { type Props } from './workflow-queries-tile.types';
+
+export default function WorkflowQueriesTile(props: Props) {
+  return (
+    <styled.Tile
+      onClick={() => props.onSelect()}
+      $isSelected={props.isSelected}
+    >
+      <styled.Header>
+        <styled.Label>
+          {props.name}
+          {/* TODO: add status icon */}
+        </styled.Label>
+        <styled.Actions>
+          <Button
+            size={SIZE.mini}
+            kind={KIND.tertiary}
+            onClick={() =>
+              props.setInput(props.input !== undefined ? undefined : '')
+            }
+          >
+            {props.input !== undefined ? 'Remove input' : 'Add input'}
```

Comment:
Adding the remove button inside the text box was not possible with the Textarea component

Reply:
It wont be added inside the text area, it placed over it.
Anyways i like replacing the add more, can you confirm with Niels.
Nit: The new approach would also need preserving the size of the button using minWidth to prevent text overflow change while moving between the two states.

Reply:
So, to accommodate longer query names?

Reply:
No, but I can give it a try
I think the button will maintain its size but let's see

Reply:
Made the designs responsive so that they work on mobile as well